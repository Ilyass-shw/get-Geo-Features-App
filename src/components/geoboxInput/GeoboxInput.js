import { FormError, Input, Label, InputField } from "./GeoBoxInputStyles";

const GeoboxInput = ({ register, errors, name, Min, Max }) => {
  return (
    <>
      <InputField>
        <Label htmlFor={name}>{name}</Label>
        <Input
          type="number"
          step="any"
          {...register(name, {
            required: "This is required.",
            min: {
              value: Min,
              message: `choose a number no less than ${Min}`,
            },
            max: {
              value: Max,
              message: `choose a number no more than ${Max}`,
            },
          })}
          Error={errors[name]}
          data-testid={name}
        />
      </InputField>
      <FormError>{errors[name] && errors[name].message}</FormError>
    </>
  );
};

export default GeoboxInput;
