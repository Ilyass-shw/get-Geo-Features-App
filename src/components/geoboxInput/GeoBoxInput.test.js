import { render } from "@testing-library/react";
import GeoboxInput from "./GeoboxInput";

describe("FeaturesCard", () => {
  const register = jest.fn();

  it("should render", () => {
    render(
      <GeoboxInput
        register={register}
        errors={{ errors: "bom" }}
        name={"randomName"}
        Min={4}
        Max={7}
      />
    );
  });
  it("should match snapshot", () => {
    const { asFragment } = render(
      <GeoboxInput
        register={register}
        errors={{ errors: "bom" }}
        name={"randomName"}
        Min={4}
        Max={7}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
