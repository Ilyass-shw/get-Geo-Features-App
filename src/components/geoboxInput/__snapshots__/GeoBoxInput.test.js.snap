// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturesCard should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL ilFTvE"
  >
    <label
      class="sc-dkPtRN hixtpS"
      for="randomName"
    >
      randomName
    </label>
    <input
      class="sc-gsDKAQ edMpyo"
      data-testid="randomName"
      step="any"
      type="number"
    />
  </div>
  <p
    class="sc-hKwDye uccYi"
  />
</DocumentFragment>
`;
